src/develop/develop.c

```diff
@@ -121,6 +126,12 @@ void dt_dev_init(dt_develop_t *dev, int32_t gui_attached)
   dev->overexposed.upper = dt_conf_get_float("darkroom/ui/overexposed/upper");
 }
 
+static void _free_form(void *data)
```

Comment:
Why do you need this helper function?
It's definition is completely equivalent to the one of `dt_masks_free_form()` - 
```
void dt_masks_free_form(dt_masks_form_t *form);
```
`dt_masks_form_t` or `void` does not matter, and casting is unneeded in C.